jgrapht-core/src/main/java/org/jgrapht/traverse/CrossComponentIterator.java

```diff
@@ -55,12 +56,18 @@
     /**
      * Iterator which provides start vertices for cross-component iteration.
      */
+    private Iterator<V> entireGraphVertexIterator = null;
```

Comment:
ok, I now understand your point. I considered the `Iterable<V>` case also a cross component iteration, while you do not. 

 Could you change the `graph.containsVertex()` check to happen lazily inside the `hasNext()` instead of the constructor?